src/Endatix.Api/Infrastructure/Cors/EndpointsCorsConfigurator.cs

```diff
@@ -0,0 +1,163 @@
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Endatix.Api.Infrastructure.Cors;
+
+public class EndpointsCorsConfigurator(IOptions<CorsSettings> corsSettings, ILogger<EndpointsCorsConfigurator> logger, IAppEnvironment appEnvironment, IWildcardSearcher wildcardSearcher) : IConfigureOptions<CorsOptions>
+{
+    public const string ALLOW_ALL_POLICY_NAME = "AllowAll";
+
+    public const string DISALLOW_ALL_POLICY_NAME = "DisallowAll";
+
+    private readonly string[] _no_input_params = [];
+
+    public void Configure(CorsOptions options)
+    {
+        AddPredefinedPolicies(options);
+
+        var settingsValue = corsSettings?.Value;
+        var isDefaultPolicySet = false;
+
+        if (settingsValue?.CorsPolicies is { Count: > 0 } corsPolicies)
+        {
+            foreach (var policySetting in corsPolicies)
+            {
+                options.AddPolicy(
+                    policySetting.PolicyName,
+                    policy => BuildFrom(policy, policySetting)
+                );
+
+                if (settingsValue?.DefaultPolicyName?.Equals(policySetting.PolicyName) == true)
+                {
+                    options.DefaultPolicyName = policySetting.PolicyName;
+                    isDefaultPolicySet = true;
+                }
+            }
+
+            if (!isDefaultPolicySet)
+            {
+                options.DefaultPolicyName = corsPolicies[0].PolicyName;
+                isDefaultPolicySet = true;
+            }
+        }
+
+        if (!isDefaultPolicySet)
+        {
+            var isDevelopment = appEnvironment.IsDevelopment();
+            options.DefaultPolicyName = isDevelopment ? ALLOW_ALL_POLICY_NAME : DISALLOW_ALL_POLICY_NAME;
+            isDefaultPolicySet = true;
+        }
+
+        logger.LogDebug("Default Cors Policy is: {@policy}", options.GetPolicy(options.DefaultPolicyName));
+    }
+
+    private CorsPolicyBuilder BuildFrom(CorsPolicyBuilder builder, CorsPolicySetting policySetting)
+    {
+        AddAllowedHeaders(builder, policySetting.AllowedHeaders);
+        AddAllowedMethods(builder, policySetting.AllowedMethods);
+        var wildcardSearchResult = AddAllowedOrigins(builder, policySetting.AllowedOrigins);
+
+        var includeAnyOrigin = wildcardSearchResult == CorsWildcardResult.MatchAll;
+        if (includeAnyOrigin && policySetting.AllowCredentials)
+        {
+            logger.LogWarning("Ignoring {setting} and disallowing credentials. Details: {details}", "AllowCredentials", "The CORS protocol does not allow specifying a wildcard (any) origin and credentials at the same time. Configure the CORS policy by listing individual origins if credentials needs to be supported.");
+            _ = builder.DisallowCredentials();
+            return builder;
+        }
+
+        _ = policySetting.AllowCredentials ? builder.AllowCredentials() : builder.DisallowCredentials();

```

Comment:
I see discarding the result many times here. I believe here the idea is that it fits in one line. Still having
```
        if (policySetting.AllowCredentials)
        {
            builder.AllowCredentials();
        }
        else
        {
            builder.DisallowCredentials();
        }
```
seems a bit more readable. Of course, this is just an opinion and both will work.

Something else, that I actually don't understand is lines (we have several such) like
`_ = builder.DisallowCredentials();`
Isn't this the same as `builder.DisallowCredentials();` and if not, what do I miss?

Reply:
The idea of discards is to only to indicate to the reader and the compiler that we don't intend to use the return value from the `builder.AllowCredentials()` or the `builder.DisallowCredentials()` methods calls. I agree that it can make reading the code harder especially to developers not used to it. I will remove it.

The single line assignment via the the ternary conditional operator is very common & clean imo. I am good both ways, I change it for traditional if/else